File: src/main/java/com/sotasan/decompiler/views/StartView.java
package com.sotasan.decompiler.views;

import com.formdev.flatlaf.extras.components.FlatLabel;
import com.sotasan.decompiler.services.LanguageService;
import lombok.Getter;
import net.miginfocom.swing.MigLayout;
import javax.swing.*;
import java.awt.event.KeyEvent;

@Getter
public class StartView extends JPanel {

    private final JPanel root;
    private final FlatLabel header;
    private final FlatLabel open;
    private final FlatLabel drag;

    public StartView() {
        setLayout(new MigLayout("fill"));

        root = new JPanel();
        root.setLayout(new MigLayout("gapy 15"));
        add(root, "center");

        header = new FlatLabel();
        header.setStyleClass("h1");
        header.setText(LanguageService.getTranslation("empty"));
        root.add(header, "wrap");

        String group = LanguageService.getTranslation("file");
        String item = LanguageService.getTranslation("file.openFile");
        String modifier = KeyEvent.getModifiersExText(KeyEvent.CTRL_DOWN_MASK);
        String key = KeyEvent.getKeyText(KeyEvent.VK_O);
        open = new FlatLabel();
        open.setText(String.format("%s > %s (%s+%s)", group, item, modifier, key));
        root.add(open, "wrap");

        drag = new FlatLabel();
        drag.setText(LanguageService.getTranslation("empty.drag"));
        root.add(drag, "wrap");
    }

}

File: jvm/src/main/kotlin/moe/sota/decompiler/jvm/views/StartView.kt
package moe.sota.decompiler.jvm.views

import com.formdev.flatlaf.extras.components.FlatLabel
import java.awt.Toolkit
import java.awt.event.KeyEvent
import javax.swing.JPanel
import moe.sota.decompiler.jvm.services.LanguageService
import net.miginfocom.swing.MigLayout

class StartView(languageService: LanguageService) : JPanel() {
    private val root: JPanel
    private val header: FlatLabel
    private val open: FlatLabel
    private val drag: FlatLabel

    init {
        layout = MigLayout("fill")

        root = JPanel().apply { layout = MigLayout("gapy 15") }
        add(root, "center")

        header =
            FlatLabel().apply {
                styleClass = "h1"
                text = languageService.getString("empty")
            }
        root.add(header, "wrap")

        open =
            FlatLabel().apply {
                val group = languageService.getString("file")
                val item = languageService.getString("file.openFile")
                val modifier =
                    KeyEvent.getModifiersExText(
                        Toolkit.getDefaultToolkit().getMenuShortcutKeyMaskEx()
                    )
                val key = KeyEvent.getKeyText(KeyEvent.VK_O)
                text = "$group > $item ($modifier + $key)"
            }
        root.add(open, "wrap")

        drag = FlatLabel().apply { text = languageService.getString("empty.drag") }
        root.add(drag, "wrap")
    }
}
